Woof/WoofTests/ViewModel/LoginViewModelTests/LoginViewModelSetUserRoleTests.swift

```diff
@@ -0,0 +1,71 @@
+import XCTest
+
+final class LoginViewModelSetUserRoleTests: XCTestCase {
+    private var userPreferencesStorage = KeyValueStorage.testPreferencesStorage
+    private let testLoginViewModel = LoginViewModel()
+
+    override func setUp() {
+        super.setUp()
+        userPreferencesStorage.deleteData(for: KeyValueStorage.Key.userPreferences)
+    }
+
+    func testMethodExistsInAPI() {
+        testLoginViewModel.set(.sitter)
+    }
+
+    func testSuccessfullyOverwritesTheDefaultValueForTheOwnerRole() {
+        // Given
+        let initialRole = Role.none
+        let overriddenRole = Role.owner
+        testLoginViewModel.set(initialRole)
+
+        // When
+        testLoginViewModel.set(overriddenRole)
+
+        // Then
+        XCTAssertNotEqual(PreferencesHandler.getUserRole(), initialRole)
+        XCTAssertEqual(PreferencesHandler.getUserRole(), overriddenRole)
+    }
+
+    func testSuccessfullyOverwritesTheDefaultValueForTheSitterRole() {
+        // Given
+        let initialRole = Role.none
+        let overriddenRole = Role.sitter
+        testLoginViewModel.set(initialRole)
+
+        // When
+        testLoginViewModel.set(overriddenRole)
+
+        // Then
+        XCTAssertNotEqual(PreferencesHandler.getUserRole(), initialRole)
+        XCTAssertEqual(PreferencesHandler.getUserRole(), overriddenRole)
+    }
+
+    func testSuccessfullyOverwritesTheSitterRoleValueForDefault() {
+        // Given
+        let initialRole = Role.sitter
+        let overriddenRole = Role.none
+        testLoginViewModel.set(initialRole)
+
+        // When
+        testLoginViewModel.set(overriddenRole)
+
+        // Then
+        XCTAssertNotEqual(PreferencesHandler.getUserRole(), initialRole)
+        XCTAssertEqual(PreferencesHandler.getUserRole(), overriddenRole)
+    }
+
+    func testReturnsExpectedValueWhenOverridesThePreviouslySavedRole() {
```

Comment:
Could you please explain what's the difference between `testSuccessfullyOverwritesTheSitterRoleValueForDefault` and `testReturnsExpectedValueWhenOverridesThePreviouslySavedRole` tests?

Reply:
There is a difference only in the enam cases that are used internally, the behavior of the test itself is the same. 
I will delete the last test.

Reply:
What are "the enam cases"?

Reply:
sry, i wanted to write "enum"